### ISSUE TITLE
Unable to Open Repository in Git Bash Shell

### ISSUE THREAD

**Issue by Jawshy** [desktop/desktop#3186]
### Description

Clicking _Repository_ → _Open in Git Bash_ does not launch a new Git Bash window set to the repository's directory. This used to work for me in the past. 

The shell is Git Bash in _File_ → _Options..._ → _Advanced_. If Git Bash is changed to Command Prompt or PowerShell then _Repository_ → _Open in [Command Prompt/PowerShell]_ will correctly work. 

### Version

**GitHub Desktop version:** 1.0.6

**OS version:** Microsoft Windows [Version 10.0.15063]

### Steps to Reproduce

1. Launch _GitHub Desktop_. 
2. Select a repository or add one if there are none listed. 
3. Click _File_ → _Options..._ → _Advanced_. Under _Shell_ choose _Git Bash_ and save. 
4. Click _Repository_ → _Open in Git Bash_. Note that nothing happens. 
5. Repeat step 3 but under _Shell_ choose _Command Prompt_ or _PowerShell_ and save. 
6. Repeat step 4. Note that the chosen shell will launch. 

**Expected behavior:** Clicking _Repository_ → _Open in Git Bash_ would launch a new Git Bash window set to the repository's directory.

**Actual behavior:** Clicking _Repository_ → _Open in Git Bash_ does nothing. 

**Reproduces how often:** 100%

### Logs

There does not appear to be any relevant information in the log files generated by GitHub
Desktop. 

#### Additional Information

I tried reinstalling both Git for Windows and GitHub Desktop but the issue persisted. The issue was also occurring on GitHub Desktop 1.0.5. I also contacted GitHub support regarding this issue.

**Comment by RosieCode95** [comment]
can confirm i have the same issue, last time i known this to work was `1.0.4`, seams to work with other shells just not `git bash` 

```
GitHub Desktop version: 1.0.6
OS version: Microsoft Windows [Version 10.0.15063 Build 15063]
```

**Comment by Elioty** [comment]
Same here!! My GitHub Desktop was on version 1.0.4 just minutes ago and it could open git bash without a problem. It told me there was a new update so I ask it to restart straight away (because I needed the new git version introduced in 1.0.5) and since then, it won't let me open git bash from it anymore...

However, when I go to the directory where git is installed, I can see it's still on version 2.14.1 and hasn't been updated to 2.14.3 like the changelog for 1.0.5 states. It might be the source of the problem. And the update process didn't ask for privilege escalation so it definitely cannot update git anyway (since it's installed in Program Files). Anyone knows how to manually trigger the git update?

**Comment by shiftkey** [comment]
@EliotyI tested this again on a development VM and found that it wasn't showing the menu entry either, so I went about recreating it.

Here's my setup:

 - Windows 10 (build `10.0.16299.19`)
 - GitHub Desktop 1.0.6
 - Git for Windows 2.14.3

I opened up regedit and looked at `HKEY_LOCAL_MACHINE\SOFTWARE\GitForWindows` - where Git for Windows registers entries for third party apps to integrate - and it has these values (as expected):

<img width="561" src="https://user-images.githubusercontent.com/359239/32148913-35111a20-bd51-11e7-929b-c9da4facfadc.png">

And these values are created as part of the [installer](https://github.com/git-for-windows/build-extra/blob/ea7ea338a5c86a1d2bd3a70582157abac93ea129/installer/install.iss#L137-L144) - and they haven't been touched since they were added ~2 years ago.

I couldn't see anything in the install log about it failing to create the registry entry (you can get this by running the installer from the command line with an additional `/log` to generate this file under `%TEMP%`) but I think that might be the problem here.

@Jawshy @RosieCode95 @Elioty can you open the registry and browse to `HKEY_LOCAL_MACHINE\SOFTWARE\GitForWindows` and confirm these entries are not present?

**Comment by Elioty** [comment]
Hello @shiftkey

Firstly my setup:
- Windows 10 (build 15063.674, stable branch, Creator Update, haven't got the Fall update yet)
- GitHub Desktop 1.0.6
- Git for Windows 2.14.3

Sadly, I have to tell you I have those entries in my registry under `HKEY_LOCAL_MACHINE\SOFTWARE\GitForWindows` (just copied it from the regedit address bar).
![capture](https://user-images.githubusercontent.com/5834064/32149114-4086f0f6-bd00-11e7-80eb-2bd63f1f7fdb.PNG)

But you said:
> I [...] found that it wasn't showing the menu entry either

The menu entry `Repository > Open in Git Bash` (if that's the one you mean) has always been there. It just does nothing when I click it.

Looking forwards to keep on helping you out hammering that issue.

**Comment by shiftkey** [comment]
@Elioty aha, that's good to know. And yes, I'm also seeing the same thing when launching. I'll let you know if I need anything from you.

**Comment by Elioty** [comment]
> And yes, I'm also seeing the same thing

So I guess the issue doesn't need the `needs-reproduction` label anymore but the `bug` label instead.
And just ping me if you need me to try something out 😉

**Comment by Jawshy** [comment]
Hi, @shiftkey. Those four entries are present in `HKEY_LOCAL_MACHINE\SOFTWARE\GitForWindows` and are identical to what you posted.

**Comment by shiftkey** [comment]
@Jawshy cool. The issue will be fixed with #3194, just wanted a sanity check that it wasn't there.

**Comment by Elioty** [comment]
Yay so the fix will make it to 1.0.7 release I guess 😄 That's really quick user support! Keep up the good work 😉

**Comment by joshaber** [comment]
>Yay so the fix will make it to 1.0.7 release I guess 

Yup! That's exactly right.
